Content.Client/GameObjects/Components/Items/HandsComponent.cs

```diff
@@ -162,20 +164,26 @@ private void UpdateHandSprites(Hand hand)
                 return;
             }
 
-            if (!entity.TryGetComponent(out ItemComponent? item)) return;
-
-            var maybeInHands = item.GetInHandStateInfo(hand.Location);
+            if (!entity.TryGetComponent(out SharedItemComponent? item))
+                return;
 
-            if (!maybeInHands.HasValue)
+            if (item.RsiPath == null)
             {
                 _sprite.LayerSetVisible($"hand-{name}", false);
             }
             else
             {
-                var (rsi, state, color) = maybeInHands.Value;
+                var rsi = IoCManager.Resolve<IResourceCache>().GetResource<RSIResource>(SharedSpriteComponent.TextureRoot / item.RsiPath).RSI;
+
+                var handName = hand.Location.ToString().ToLowerInvariant();
+                var prefix = item.EquippedPrefix;
+                var state = prefix != null ? $"{prefix}-inhand-{handName}" : $"inhand-{handName}";
+
+                var color = item.Color;
+
                 _sprite.LayerSetColor($"hand-{name}", color);
                 _sprite.LayerSetVisible($"hand-{name}", true);
-                _sprite.LayerSetState($"hand-{name}", state, rsi);
+                _sprite.LayerSetState($"hand-{name}", state, rsi);  
```

Comment:
this should be a visualizer.

If you also want to get fancy we need support for multiple layers in-hands (including unshaded layers).

Reply:
Will handle in future hand refactor PRs.